src/main/java/org/apache/commons/csv/CSVFormat.java

```diff
@@ -2280,6 +2279,27 @@ private void validate() throws IllegalArgumentException {
         }
     }
 
+    private Object readResolve() throws ObjectStreamException {
+        // check whether field duplicateHeaderMode (as of 1.10.0) was found in serialized form
+        // if not, set from boolean allowDuplicateHeaderNames
+        if (duplicateHeaderMode == null) {
+            setFieldValue("duplicateHeaderMode",
+                    allowDuplicateHeaderNames ? DuplicateHeaderMode.ALLOW_ALL : DuplicateHeaderMode.DISALLOW);
+        }
+        return this;
+    }
+
+    private void setFieldValue(final String fieldName, final Object value) {
+        try {
+            Field fld = getClass().getDeclaredField(fieldName);
+            fld.setAccessible(true); // make field non-final
+            fld.set(this, value);
+            fld.setAccessible(false);
+        } catch (Exception ex) {

```

Comment:
Unlikely to need to catch `Exception`, catching `ReflectiveOperationException` is cleaner IMO. But where is this used from anyway? If it's only called from `readResolve()` why not turn it into an `ObjectStreamException`?

Reply:
The code in `try` might throw any of `NoSuchFieldException, SecurityException, IllegalArgumentException, IllegalAccessException`
`ObjectStreamException` is abstract, but a subclass can be used. `InvalidClassException` seems like the good choice.